.github/workflows/tests.yml

```diff
@@ -60,3 +60,163 @@ jobs:
           flags: tests
           env_vars: PYTHON
           name: ${{ matrix.python }}
+
+  cygwin:
+    runs-on: windows-latest
+    env:
+      FORCE_COLOR: true
+    strategy:
+      fail-fast: false
+      matrix:
+        python:
+          - '3.9'
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Cygwin
+        uses: cygwin/cygwin-install-action@v2
+        with:
+          packages: >-
+            python39
+            python39-devel
+            python39-pip
+            python39-setuptools
+            cmake
+            gcc-core
+            gcc-g++
+            git
+            make
+            ninja
+
+      - name: Fix git dubious ownership
+        # This addresses the "fatal: detected dubious ownership in
+        # repository" and "fatal: not in a git directory" errors
+        # encountered when trying to run Cygwin git in a directory not
+        # owned by the current user. This happens when the tests run
+        # Cygwin git in a directory outside the Cygwin filesystem.
+        run: git config --global --add safe.directory '*'
+        shell: C:\cygwin\bin\env.exe CYGWIN_NOWINPATH=1 CHERE_INVOKING=1 C:\cygwin\bin\bash.exe -leo pipefail -o igncr {0}
+
+      - name: Get pip cache path
+        id: pip-cache-path
+        run: echo "path=$(cygpath -w $(python -m pip cache dir))" >> $GITHUB_OUTPUT
+        shell: C:\cygwin\bin\env.exe CYGWIN_NOWINPATH=1 CHERE_INVOKING=1 C:\cygwin\bin\bash.exe -leo pipefail -o igncr {0}
+
+      - name: Restore cache
+        # Cygwin Python cannot use binary wheels from PyPI. Building
+        # some dependencies takes considerable time. Caching the built
+        # wheels speeds up the CI job quite a bit.
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.pip-cache-path.outputs.path }}
+          key: cygwin-pip-${{ github.sha }}
+          restore-keys: cygwin-pip-
+
+      - name: Install
+        run: python -m pip --disable-pip-version-check install .[test]
```

Comment:
The nox configuration also installs `pytest-github-actions-annotate-failures`; do you wish to install that as well?

Reply:
I dislike stuff that annotates the diffs on GitHub it makes reviewing them a pain. But I can reinstate this one if people like it.

Reply:
If it just lets you jump to the start of the detailed error reports in the CI logs, I can see the use for that; if it mostly clutters the diff that's less useful.

I just found it as a side effect of checking that the tests didn't use any of the `GITHUB_*` environment variables, since I remembered seeing one recently.